### PR TITLE
Release v2, update dependencies, remove neo-utils-go 

### PR DESCRIPTION
**Pull request by epavlova**
# Description

## What

Release v2 of the library with updated dependencies. The dependency to neo-utils-go is removed by removing a specific error type defined in neo-utils-go. However, in practice `neoutils.ConstraintViolationError` was never returned by neo-utils-go so we are safe to remove it.

## Why

https://financialtimes.atlassian.net/browse/UPPSF-2893

## Anything, in particular, you'd like to highlight to reviewers

_New major version is released mostly because when we move to modules, we create new major versions of the libraries._

## Scope and particulars of this PR (Please tick all that apply)

- [X] Tech hygiene (dependency updating & other tech debt)
- [ ] Bug fix
- [ ] Feature
- [ ] Documentation
- [X] Breaking change
- [ ] Minor change (e.g. fixing a typo, adding config)

___
This Pull Request follows the rules described in our [Pull Requests Guide](https://github.com/Financial-Times/upp-docs/tree/master/guides/pr-guide)

### CODE REVIEW

**Review by ivan-p-nikolov — APPROVED** [review]

File: baseftrwapp/http_handlers.go [hunk 1]
```diff
@@ -58,12 +57,6 @@ func (hh *httpHandlers) putHandler(w http.ResponseWriter, req *http.Request) {
...
 			return
-		case *neoutils.ConstraintViolationError:
-			// TODO: remove neo specific error check once all apps are
-			// updated to use neoutils.Connect() because that maps errors
-			// to rwapi.ConstraintOrTransactionError
-			writeJSONMessage(w, e.Error(), http.StatusConflict)
-			return
 		case rwapi.ConstraintOrTransactionError:
 			writeJSONMessage(w, e.Error(), http.StatusConflict)
 			return
```
Comment (ivanruski) [inline baseftrwapp/http_handlers.go:62]: **question:** Does this mean that we have to start returning `rwapi.ConstraintOrTransactionError` in our services ?

**Review by ivanruski — APPROVED** [review]

File: baseftrwapp/http_handlers.go (hunk 1 above)
Comment (epavlova) [reply baseftrwapp/http_handlers.go:62]: Agh, yes. This is an interesting one. I think you are right, in practice we are never reaching that case(`case rwapi.ConstraintOrTransactionError:`). It's up to us if we want to utilise that case and error and even whether we still want to utilise `base-ft-rw-app-go` in new projects. So far we are supporting it but we are not using it when building new services.
Looking at the following search in GitHub `org:Financial-Times rwapi.ConstraintOrTransactionError`, I can see that the error is in use in a few deprecated projects and a few active ones. The active projects are:
- Financial-Times/neo-utils-go, which we plan to deprecate;
- Financial-Times/concepts-rw-neo4j where the error is used [in a way](https://github.com/Financial-Times/concepts-rw-neo4j/blob/bdfe8608a1ec5d924dd981f7b0b9473b245c2dbc/concepts/handlers.go#L75) that has nothing to do with `base-ft-rw-app-go` but with `neo-utils-go`. The new driver does not return such error so effectively that code in `concepts-rw-neo4j` is dead and most probably we don't need it;
CC @ivan-p-nikolov @ivanruski

**Comment by epavlova** [comment]
I'm merging the PR as the conversation above won't change this PR.

File: baseftrwapp/http_handlers.go (hunk 1 above)
Comment (ivan-p-nikolov) [reply baseftrwapp/http_handlers.go:62]: Previous team had some idea behind `base-ft-rw-app-go`, but without that context I don't see the value in it. For me it only makes the code harder to read. As you can tell I'm all for deprecating this functionality.

**Pull request merged**